libs/contrib/Data/HVect.idr

```diff
@@ -115,20 +115,56 @@ public export
   show xs = "[" ++ (pack . intercalate [','] . map unpack . toList $ shows xs) ++ "]"
 
 ||| Extract an arbitrary element of the correct type.
-||| @ t the goal type
+|||
+||| ```idris example
+||| > get [1, "string"] {p = Here}
+||| 1
+||| ```
 public export
 get : HVect ts -> {auto 1 p : Elem t ts} -> t
 get (x :: xs) {p = Here} = x
 get (x :: xs) {p = (There p')} = get xs
 
-||| Replace an element with the correct type.
+||| Replace an element with the correct type. (Homogeneous)
+|||
+||| ```idris example
+||| > put 2 [1, "string"]
+||| [2, "string"]
+||| ```
 public export
 put : t -> HVect ts -> {auto 1 p : Elem t ts} -> HVect ts
 put y (x :: xs) {p = Here} = y :: xs
 put y (x :: xs) {p = (There p')} = x :: put y xs
 
-||| Update an element with the correct type.
+||| Replace an element with the correct type. (Heterogeneous)
+|||
+||| ```idris example
+||| > htPut True [1, "string"] {p = Here}
+||| [True, "string"]
+||| ```
+public export
+htPut : u -> HVect ts -> {auto 1 p : Elem t ts} -> HVect (replaceByElem ts p u)
+htPut y (x :: xs) {p = Here} = y :: xs
+htPut y (x :: xs) {p = There p'} = x :: htPut y xs
+
+||| Update an element with the correct type. (Homogeneous)
+|||
+||| ```idris example
+||| > update "hello world!" [1, "string"]
+||| [1, "hello world!"]
+||| ```
+public export
+update : t -> HVect ts -> {auto 1 p : Elem t ts} -> HVect ts
```

Comment:
Note that this is just `put`. `update` should take a `t -> t` function as an argument.

Reply:
Thanks for catching this oversight, fixed!